### ISSUE TITLE
Player saturation doesn't affect their hand

### ISSUE THREAD

**Issue by immagiov4** [minetest/minetest#14169]
### Minetest version

```true
Minetest 5.8
Using Irrlicht 1.9.0mt13
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="."
STATIC_LOCALEDIR="locale"
```

### Operating system and version

Windows 11

### Summary

Well, the title says it all. Example:
![image](https://github.com/minetest/minetest/assets/55757849/856ae688-9002-419e-9c4c-fa2561511931)

### Steps to reproduce

Set the player saturation to 0 using `player:set_lighting({saturation = 0})`

**Comment by appgurueu** [comment]
FWIW, as a "workaround", you can manually desaturate the hand by setting a different hand item and using an appropriate texture (texture modifiers should work here).

(It might even be argued that this is preferable: Some games might want to desaturate the environment, but not the hand, which may be considered part of the HUD rather than part of the environment.)

**Comment by Zughy** [comment]
@appgurueu that would work with the default hand but it should break as soon as you hold an item

**Comment by appgurueu** [comment]
That's true, it'd be pretty awkward to register desaturated variants of all items.

Still, I see a potential issue if there are users who consider the wield item part of the HUD and would specifically like for it to be unaffected by desaturation (and other environmental rendering effects); think for example of a game where you're the colorful savior of a desaturated environment.

Given that 5.8 shipped, making the hand be affected by this rendering effect might even be considered a (minor) breaking change. So a flag will probably have to do.

**Comment by Zughy** [comment]
I mean, nobody has really complained so far and I'm not aware of games using such effects

**Comment by grorp** [comment]
Related: #13338
